apps/chef_objects/src/chef_key.erl

```diff
@@ -155,12 +164,28 @@ parse_binary_json(Bin, undefined) ->
     chef_object_base:validate_ejson(EJ, chef_object_base:public_key_spec(req)),
 
     % validate name field
-    chef_object_base:validate_ejson(EJ, {[ {<<"name">>, {string_match, chef_regex:regex_for(key_name)}},
-                                           {{req, <<"expiration_date">>}, string} ]}),
-
-    %% this will raise if expiration_date isn't a valid datetime
-    parse_expiration(ej:get({<<"expiration_date">>}, EJ)),
-    EJ.
+    chef_object_base:validate_ejson(EJ, {[
+                                           {<<"name">>, {string_match, chef_regex:regex_for(key_name)}}
+                                         ]}),
+
+    % validate that the expiration_date field is a ISO8601 UTC timestring (ending in Z) and that
+    % ec_date can parse it, and then turn it into a format sqerl's deps can handle
+    try
+        chef_object_base:validate_ejson(EJ, {[
+                                              {{req, <<"expiration_date">>},
+                                               {string_match, chef_regex:regex_for(date)}}
+                                             ]}),
+        parse_expiration(ej:get({<<"expiration_date">>}, EJ)),
+
+        % remove the Z from the end of the timestring so sqerl doesn't blow up on insert
+        SafeTimestring = oc_chef_wm_base:convert_input_utc_timestring_into_db_parseable_format(ej:get({<<"expiration_date">>}, EJ)),
```

Comment:
Sure, there is a lot of stuff I wanted to put into oc_chef_wm_base, but how do you know if a field is a date? That is why I'm doing it here.